### PR TITLE
Cross tenant app denial

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
This resolves issues with cross tenant apps, specifically when interacting with public apps.

addresses: https://linear.app/budibase/issue/BUDI-7811/vulnerability-cross-tenant-cookies

### CODE REVIEW

**Review by aptkingston — APPROVED** [review]
LGTM!

File: packages/server/src/middleware/currentapp.ts [hunk 1]
```diff
@@ -69,28 +70,34 @@ export default async (ctx: UserCtx, next: any) => {
...
-      !tenancy.isUserInAppTenant(requestAppId, ctx.user)
-    ) {
-      // don't error, simply remove the users rights (they are a public user)
-      ctx.user = users.cleanseUserObject(ctx.user) as ContextUser
-      ctx.isAuthenticated = false
-      roleId = roles.BUILTIN_ROLE_IDS.PUBLIC
-    }
+  const userId = ctx.user ? generateUserMetadataID(ctx.user._id!) : undefined
 
+  // if the user not in the right tenant then make to wipe their cookie
```
Comment (samwho) [inline packages/server/src/middleware/currentapp.ts:75]:
```suggestion
  // if the user is not in the right tenant then make sure to wipe their cookie
```

Bit of a nit but 👼

**Review by samwho — COMMENTED** [review]
Is there a way to add tests for this?

Also, do we need to worry about any of the other `TenantResolutionStrategy` types?

**Comment by mike12345567** [comment]
> Is there a way to add tests for this?
> 
> Also, do we need to worry about any of the other `TenantResolutionStrategy` types?

Testing this could be quite difficult, we don't really have a test setup for this right now, but its probably one worth putting with QA-wolf!

The other tenant strategies aren't affected, this was just a problem with local debugging, won't be a production issue.

File: packages/backend-core/src/tenancy/tenancy.ts [hunk 2]
```diff
@@ -93,11 +93,19 @@ export const getTenantIDFromCtx = (
   // subdomain
   if (isAllowed(TenantResolutionStrategy.SUBDOMAIN)) {
     // e.g. budibase.app or local.com:10000
-    const platformHost = new URL(getPlatformURL()).host.split(":")[0]
+    let platformHost
+    try {
+      platformHost = new URL(getPlatformURL()).host.split(":")[0]
+    } catch (err: any) {
+      // if invalid URL, just don't try to process subdomain
+      if (err.code !== "ERR_INVALID_URL") {
```
Comment (adrinr) [inline packages/backend-core/src/tenancy/tenancy.ts:101]: In which case we want to support invalid URLs?

Comment (mike12345567) [reply packages/backend-core/src/tenancy/tenancy.ts:101]: Just local development, it won't happen in multi-tenant environments, but this was blocking the ability to fetch tenant IDs in other ways

**Pull request merged**